packages/sdk-bridge/src/backend.ts

```diff
@@ -0,0 +1,242 @@
+import {  GoldSkyBackend, GoldSkyMessage, MessageFilter } from "@nomad-xyz/sdk";
+import {MessageBackend} from "@nomad-xyz/sdk";
+import { request, gql } from 'graphql-request';
+import * as config from '@nomad-xyz/configuration';
+
+const defaultGoldSkySecret = "";
+
+
+/**
+ * Abstract class required for operation of NomadMessage
+ */
+ export default abstract class BridgeMessageBackend extends MessageBackend {
+    abstract sender(messageHash: string): Promise<string | undefined>;
+    abstract receivedTx(messageHash: string): Promise<string | undefined>;
+}
+
+
+export type GoldSkyBridgeMessage =  GoldSkyMessage & {
+    send_tx: string,
+    sent_at: string,
+    send_block: string,
+    original_sender: string,
+    receive_tx: string,
+    origin_and_nonce: string,
+    receive_block: string,
+    received_at: string,
+};
+
+/**
+ * GoldSky bridge backend for NomadMessage 
+ */
+export class GoldSkyBridgeBackend extends GoldSkyBackend implements BridgeMessageBackend {
+
+    messageCache: Map<string, GoldSkyBridgeMessage>;
+
+    constructor(env: string, secret: string) {
+        super(env, secret);
+        this.messageCache = new Map();
+    }
+
+    /**
+     * Checks whether an environment is supported by the backend. Throws on unsupported
+     * @param environment environment to check
+     */
+    static checkEnvironment(environment: string): void {
+        if (environment != 'production') {
+            throw new Error(`Only production environment is supported`);
```

Comment:
Will need to update to support staging

Reply:
Yep, waiting it to be deployed, and then also goldsky need to support it